### PR TITLE
Add batched version of trtrs

### PR DESCRIPTION
**Pull request by vishwakftw**
- Remove single batch TH/THC implementations
- Remove `_batch_trtrs_lower` from `multivariate_normal`
- Add tests for batched behavior
- Modify trtrs_backward to accommodate for batched case
- Modify docs

In a future PR, this will be renamed to `triangular_solve`.

### CODE REVIEW

File: aten/src/ATen/native/BatchLinearAlgebra.cpp [hunk 1]
```diff
@@ -320,7 +339,9 @@ Tensor& cholesky_solve_out(Tensor& result, const Tensor& self, const Tensor& A,
   AT_CHECK(self.dim() == 2 && A.dim() == 2,
            "torch.cholesky_solve() with the `out` keyword does not support batching. "
            "b.dim() (", self.dim(), ") and A.dim() (", A.dim(), ") must both be 2.");
-  result = at::_cholesky_solve_helper(self, A, upper);
+  Tensor result_tmp;
+  result_tmp = at::_cholesky_solve_helper(self, A, upper);
+  result.resize_as_(result_tmp).copy_(result_tmp);
   return result;
```
Comment (ifedan) [inline aten/src/ATen/native/BatchLinearAlgebra.cpp]: Nice catch!

File: aten/src/ATen/native/BatchLinearAlgebra.cpp [hunk 2]
```diff
@@ -621,4 +644,79 @@ Tensor& triu_cpu_out(Tensor &result, const Tensor& self, int64_t k) {
...
+    singleCheckErrors(infos[0], "trtrs_cpu");
+  }
+  return std::tuple<Tensor, Tensor>(self_working_copy, A_working_copy);
+}
+
+// Supports arbitrary batch dimensions for self and A
+std::tuple<Tensor, Tensor> trtrs(const Tensor& self, const Tensor& A, bool upper, bool transpose, bool unitriangular) {
+  AT_CHECK(self.dim() >= 2,
+           "b should have at least 2 dimensions, but has ", self.dim(), " dimensions instead");
+  AT_CHECK(A.dim() >= 2,
```
Comment (ifedan) [inline aten/src/ATen/native/BatchLinearAlgebra.cpp]: Previously we accept dim = 1, do we want to exclude this case?

Comment (vishwakftw) [reply aten/src/ATen/native/BatchLinearAlgebra.cpp]: `trtrs` has the same issue as `gesv` stated in https://github.com/pytorch/pytorch/issues/15328. This case attempts to solve the issue.

**Comment by vishwakftw** [comment]
@ifedan Is this good to go?

**Comment by ifedan** [comment]
> @ifedan Is this good to go?
yep, looks good to me.

**Review by ifedan — APPROVED** [review]

**Pull request closed without merging**